app/api/get-token/__tests__/route.spec.ts

```diff
@@ -0,0 +1,72 @@
+/**
+ * @jest-environment node
+ */
+import { GET, POST, PUT, DELETE } from '../route';
+import { getAuthTokenServer } from '@/utils/getAuthTokenServer';
+
+jest.mock('@/utils/getAuthTokenServer', () => ({
+    getAuthTokenServer: jest.fn(),
+}));
+
+describe('GET Function', () => {
+    afterEach(() => {
+        jest.clearAllMocks();
+    })
+
+    it('should return token if available and a status of 200', async () => {
+        (getAuthTokenServer as jest.Mock).mockResolvedValue('valid-token');
+
+        const response = await GET();
+        expect(response.status).toEqual(200);
+        const data = await response.json();
+
+        expect(data).toEqual({ token: 'valid-token' })
+    })
+
+    it('should return value null for token if token is not available', async () => {
+        (getAuthTokenServer as jest.Mock).mockResolvedValue(null);
+
+        const response = await GET();
+        expect(response.status).toEqual(200);
+        const data = await response.json();
+
+        expect(data).toEqual({ token: null })
+    })
+
+    it('should handle internal server error', async () => {
+        (getAuthTokenServer as jest.Mock).mockRejectedValue(new Error('Test error'));
+
+        const response = await GET();
+        expect(response.status).toEqual(405);
```

Comment:
`405: Method not Allowed` doesn't seem to be the correct error code for this case. The test doesn't seem to check if the correct method (GET, POST etc.) is used.

Further down, the error message also reads as a `Internal Server Error` which should be a `500` code. So it feels like things aren't matching up here.

Reply:
Thanks @andrewebdev. The 500 error code totally makes sense to me since it's on the server side. I think I was copying and pasting and forgot to change that one.